### PR TITLE
fix: array of objects with properties

### PR DESCRIPTION
**Pull request by P0lip**

### CODE REVIEW

File: src/utils/walk.ts [hunk 1]
```diff
@@ -33,20 +33,28 @@ function assignNodeSpecificFields(base: IBaseNode, node: JSONSchema4) {
   }
 }
 
+function getType(node: JSONSchema4) {
+  if (Array.isArray(node.type)) {
+    return node.type.length === 1 ? node.type[0] : node.type;
```
Comment (lottamus) [inline src/utils/walk.ts]: We return a string if there's only one, but an array if theres none or more than one? Is that correct? Just making sure

**Comment by lottamus** [comment]
Do we have this fix on the v2 branch as well?

File: src/utils/walk.ts (hunk 1 above)
Comment (P0lip) [reply src/utils/walk.ts]: > We return a string if there's only one, but an array if theres none or more than one? Is that correct? Just making sure

Well, I'm not sure how to handle such a scenario. For instance, there is no way we can display both `array` of items and `object` with properties.
Our UI is just too simple.

We would need to have something more or less similar to http://jlblcc.github.io/json-schema-viewer/ to handle such a case.

@marbemac what your thought on that?

Comment (P0lip) [reply src/utils/walk.ts]: Also, we should be handling `not`, which we don't at the moment.
Any suggestions on the UI?

Comment (lottamus) [reply src/utils/walk.ts]: > there is no way we can display both array of items and object with properties

Is this what you mean?

image


Or something like `oneOf`?

image

Comment (marbemac) [reply src/utils/walk.ts]: What is `not`?

We don't have to handle array and object in same property. We can suggest people use `oneOf` if they need that, although its kind of crazy and definitely an edge case.

Comment (P0lip) [reply src/utils/walk.ts]: https://json-schema.org/understanding-json-schema/reference/combining.html#not

Comment (P0lip) [reply src/utils/walk.ts]: @lottamus yeah, pretty much that
```json
{
  "type": ["array", "object"],
  "properties": {
    "bar": { 
      "type": "string" 
    }
  },
  "items": [
     {
       "type": "number"
     },
     {
       "type": "object",
       "properties": {
          "foo": { 
            "type": "number" 
          }
        }
     }
  ]
}

Comment (marbemac) [reply src/utils/walk.ts]: Got it, we can skip `not` for now - I haven't seen it used in the wild by our customers yet, and would need to support it in JSE as well. A good addition for later, but not high priority right now.

**Pull request merged**